### PR TITLE
Add Cache statistics to metrics

### PR DESCRIPTION
**Pull request by bradyoo**

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java [hunk 1]
```diff
@@ -85,8 +90,17 @@ public Cache(CacheConfiguration conf, String name, MetricKey metricKey) {
     mEvictionThread = new EvictionThread();
     mEvictionThread.setDaemon(true);
     // The eviction thread is started lazily when we first reach the high water mark.
+    mStatsCounter = new StatsCounter();
 
     MetricsSystem.registerGaugeIfAbsent(metricKey.getName(), mMap::size);
+    MetricsSystem.registerGaugeIfAbsent(metricKey.getName() + "-hits",
```
Comment (LuQQiu) [inline core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java:96]: Can you help change to use a different MetricKey for each Metric?
In this way, we have a central place storing all the metric names and their descriptions.

File: core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java [hunk 2]
```diff
@@ -85,8 +90,17 @@ public Cache(CacheConfiguration conf, String name, MetricKey metricKey) {
...
+        mStatsCounter.mHitCount::get);
+    MetricsSystem.registerGaugeIfAbsent(metricKey.getName() + "-misses",
+        mStatsCounter.mEvictionCount::get);
```
Comment (LuQQiu) [inline core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java:99]: mMissCount?

**Review by LuQQiu — COMMENTED** [review]
Thanks for adding the metastore metrics!  Left some comments

**Comment by bradyoo** [comment]
Feedback applied.

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 3]
```diff
@@ -231,10 +231,38 @@ public MetricKey build() {
...
+          .setDescription("Total number of misses in the edge (inode metadata) cache. "
+              + "The edge cache is responsible for managing the mapping "
+              + "from (parentId, childName) to childId.")
+          .setMetricType(MetricType.GAUGE)
+          .build();
   public static final MetricKey MASTER_EDGE_CACHE_SIZE =
       new Builder(Name.MASTER_EDGE_CACHE_SIZE)
           .setDescription("Total number of edges (inode metadata) cached. "
-              + "The edge cache is responsible for managing the mapping "
+              + "The edge cache is responsible fosr managing the mapping "
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:265]: a typo?

**Review by LuQQiu — APPROVED** [review]
LGTM, thanks for adding the metrics!
Please fix the small typo before merging it

**Comment by bradyoo** [comment]
alluxio-bot, merge this please.

**Comment by bradyoo** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please.

**Pull request merged**